pilot/pkg/security/model/authentication.go

```diff
@@ -251,13 +251,24 @@ func ApplyCredentialSDSToServerCommonTLSContext(tlsContext *tls.CommonTlsContext
 	// create SDS config for gateway/sidecar to fetch key/cert from agent.
 	caCert := tlsOpts.CredentialName + SdsCaSuffix
 	if len(tlsOpts.CredentialNames) > 0 {
+		hasCacert := false
 		// Handle multiple certificates for RSA and ECDSA
-		tlsContext.TlsCertificateSdsSecretConfigs = make([]*tls.SdsSecretConfig, len(tlsOpts.CredentialNames))
-		for i, name := range tlsOpts.CredentialNames {
-			tlsContext.TlsCertificateSdsSecretConfigs[i] = ConstructSdsSecretConfigForCredential(name, credentialSocketExist, push)
+		tlsContext.TlsCertificateSdsSecretConfigs = []*tls.SdsSecretConfig{}
+		for _, name := range tlsOpts.CredentialNames {
+			if strings.HasSuffix(name, SdsCaSuffix) {
```

Comment:
I am worried about this. First, let me say I think our existing API is quite problematic which is not your fault of course -- but something that is widely used and that we need to keep compatibility with its behavior.

Today, a user can put `credentialName: blah-cacert` and it will be treated as an opaque string. Now, the behavior is changed and we are implicitly deriving that they mean to use this as-is.

Reply:
overloading the string content like `cacert://` wouldn't work well since we already used (abused) that to have a meaning about the type.

It does seem like a new field `caCertCredentialName` may really be appropriate. WDYT @keithmattix